src/App.tsx

```diff
@@ -15,36 +16,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Length = 'length',
+  Alphabetically = 'alphabetically',
+}
+
+const sortList = (
+  goods: string[],
+  sortBy: SortType,
+  isReversed: boolean,
+): string[] => {
+  const sortedGoods = [...goods].sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Alphabetically:
+        return a.localeCompare(b);
+      case SortType.Length:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortedBy, setSortedBy] = useState<SortType>(SortType.None);
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const sortedGoods = sortList(goodsFromServer, sortedBy, reversed);
+
+  const resetGoods = () => {
+    setSortedBy(SortType.None);
+    setReversed(false);
+  };
```

Comment:
[CODE STYLE] - Function's name should start with a verb. Consider renaming 'resetGoods' to 'handleResetGoods' to make it clear that it's an event handler.